### PR TITLE
Customize payload on jstree init (rebased onto develop)

### PR DESCRIPTION
**Pull request by atarkowska**
This is the same as gh-4900 but rebased onto develop.

---

# What this PR does

allow passing custom payload on jstree init. @will-moore suggested that trick

### CODE REVIEW

**Comment by atarkowska** [comment]
--rebased-from #4900

File: components/tools/OmeroWeb/omeroweb/webclient/static/webclient/javascript/ome.tree.js [hunk 1]
```diff
@@ -390,8 +390,11 @@ $(function() {
             'data' : function(node, callback, payload) {
                 // Get the data for this query
                 if (payload === undefined) {
-                    payload = {};
+                    payload = this.element.data('payload');
```
Comment (will-moore) [inline components/tools/OmeroWeb/omeroweb/webclient/static/webclient/javascript/ome.tree.js:393]: I think a bit more info could be handy to remind us (me) or the use-case here, E.g:
// Check for existing 'payload' data, used to initialise the jsTree

**Comment by will-moore** [comment]
What's the requirement for the `get_bool_or_default` fix?
This makes the queries a bit more relaxed. E.g. ?orphaned=t would previously be evaluated as False but is now True. Same for 'ON', 'T', 'Y' etc.
I guess this is unlikely to lead to any bugs, but I'm not sure it's expected behaviour.

**Comment by atarkowska** [comment]
becuse there is a bug

```
  File "/home/omero/OMERO.py/lib/python/omeroweb/webclient/views.py", line 148, in get_bool_or_default
    if val_raw.lower() == 'true' or int(val_raw) == 1:
ValueError: invalid literal for int() with base 10: 'false'
```

**Comment by atarkowska** [comment]
`toBoolean` is a method in `omero.gateway`. I don't see anything wrong with unifying web and python api. `on` is a valid result from checkbox input field.

**Comment by will-moore** [comment]
To retain the same behaviour and fix that bug in `get_bool_or_default()` something like this should work:

```
    val = request.GET.get(name, default)
    return str(val).lower() in ['true', '1']
```

**Comment by atarkowska** [comment]
`toBoolean` doesn't change any behaviour. It only expands for more options.  Hard-coding them has no sense to me really.

**Comment by atarkowska** [comment]
closing in fav of https://github.com/openmicroscopy/openmicroscopy/pull/4923

**Pull request closed without merging**